app/gui/src/presenter/graph/state.rs

```diff
@@ -650,6 +656,25 @@ impl<'a> ViewChange<'a> {
             None
         }
     }
+
+    /// Set the node expression.
+    pub fn set_node_expression(&self, id: ViewNodeId, expression: String) -> Option<AstNodeId> {
+        let mut nodes = self.nodes.borrow_mut();
+        let ast_id = nodes.ast_id_of_view(id)?;
+        let displayed = nodes.get_mut(ast_id)?;
+        let expression = node_view::Expression::new_plain(expression);
+        tracing::debug!(
+            "Setting node expression from view: {} -> {}",
+            displayed.expression,
+            expression
+        );
```

Comment:
For me, extracting the message makes the debug statement much less readable. 

```
let debug_message = "Setting node expression from view: {} -> {}";
tracing::debug!(debug_message, displayed.expression, expression);
```

The extra variable does not add any useful contextual information, but now when scanning the debug statement, one has to trace back to the variable to read its content.
Before, one could just read the debug statement linearly top to bottom.

Reply:
> you can make it 2 lines instead of 5 lines by extracting &str to var

I'm not sure if it has the same performance: the format macro can adjust allocated space when knows the string on compilation time.

Reply:
I believe it has the same performance - this is `&'static str`, it is not allocated, it is build-into the binary, it sohuld be:

```rust
let debug_message = "Setting node expression from view:";
tracing::debug!("{debug_message} {} -> {}", displayed.expression, expression);
```

It's sharter to look at and I think we should optimize the code lengh, otherwise it's hard to read. If we can change it, I'd be thankful.